ledger/participant-state/kvutils/src/test/suite/scala/com/daml/ledger/participant/state/kvutils/KVUtilsTransactionSpec.scala

```diff
@@ -80,6 +81,40 @@ class KVUtilsTransactionSpec extends AnyWordSpec with Matchers with Inside {
     val p0 = mkParticipantId(0)
     val p1 = mkParticipantId(1)
 
+    "add the key of an exercised contract as a submission input" in KVTest.runTestWithSimplePackage(
+      alice,
+      bob,
+      eve,
+    ) { simplePackage =>
+      val seed0 = seed(0)
+      val seed1 = seed(1)
+      for {
+        transaction1 <- runSimpleCommand(alice, seed0, simpleCreateCmd(simplePackage))
+        result <- submitTransaction(
+          submitter = alice,
+          transaction = transaction1,
+          submissionSeed = seed0,
+        )
+        (entryId, logEntry) = result
+        contractId = contractIdOfCreateTransaction(
+          KeyValueConsumption.logEntryToUpdate(entryId, logEntry)
+        )
+
+        transaction2 <- runSimpleCommand(
+          alice,
+          seed1,
+          simplePackage.simpleExerciseConsumeCmd(contractId),
+        )
+        submission <- prepareTransactionSubmission(
+          submitter = alice,
+          transaction = transaction2,
+          submissionSeed = seed1,
+        )
+      } yield {
+        submission.getInputDamlStateList.asScala.filter(_.hasContractKey) should not be empty
```

Comment:
Can you make this assertion more specific, please? I.e., it is not apparent if there needs to be 1 or more of such state keys and to which contract(s) these should be pointing to.

Reply:
I will do so as part of #8529.